rebel-readline/src/rebel_readline/core.clj

```diff
@@ -82,6 +82,52 @@
   []
   "[Rebel readline] Type :repl/help for online help info")
 
+(defn read-line-opts
+  "Like read-line, but allows overriding of the LineReader prompt, buffer, and mask parameters.
+   
+   :prompt 
+     Allows overriding with a cusom prompt
+   :buffer
+     The default value presented to the user to edit, may be null.
+   :mask 
+     Should be set to a single character used by jline to bit-mask.  
+     Characters will not be echoed if they mask to 0
+     Might do crazy stuff with rebel-readline, use with caution.
+     defaults to nil (no mask)
+   :command-executed
+     sentinal value to be returned when a repl command is executed, otherwise a 
+     blank string will be returned when a repl command is executed.
+  "
+  [ & {prompt :prompt
+       mask :mask
+       buffer :buffer 
+       command-executed :command-executed
+       :or {prompt nil buffer nil mask nil command-executed ""}}]
```

Comment:
Default opts are set up so that without any options, `(read-line-opts)` does the same thing as `(read-line)`